airbyte-server/src/main/java/io/airbyte/server/handlers/WorkspacesHandler.java

```diff
@@ -24,37 +24,141 @@
+        .withTombstone(false);
+
+    if (!Strings.isNullOrEmpty(email)) {
+      workspace.withEmail(email);
+    }
+
+    try {
+      if (getWorkspaceBySlug(workspace.getSlug()) != null) {
+        throw new KnownException(HttpStatus.CONFLICT_409, "A workspace already exists with the same name");
+      }
+    } catch (NotFoundException e) {
+      // if the workspace does not exist, we can create a new one
+    }
+
+    configRepository.writeStandardWorkspace(workspace);
+
+    return buildWorkspaceRead(workspace);
+  }
+
+  public void deleteWorkspace(final WorkspaceIdRequestBody workspaceIdRequestBody)
+      throws JsonValidationException, IOException, ConfigNotFoundException {
+    // get existing implementation
+    final StandardWorkspace persistedWorkspace = configRepository.getStandardWorkspace(workspaceIdRequestBody.getWorkspaceId());
+
+    // disable all connections associated with this workspace
+    for (ConnectionRead connectionRead : connectionsHandler.listConnectionsForWorkspace(workspaceIdRequestBody).getConnections()) {
+      connectionsHandler.deleteConnection(connectionRead);
+    }
+
+    // disable all destinations associated with this workspace
+    for (DestinationRead destinationRead : destinationHandler.listDestinationsForWorkspace(workspaceIdRequestBody).getDestinations()) {
+      destinationHandler.deleteDestination(destinationRead);
+    }
+
+    // disable all sources associated with this workspace
+    for (SourceRead sourceRead : sourceHandler.listSourcesForWorkspace(workspaceIdRequestBody).getSources()) {
+      sourceHandler.deleteSource(sourceRead);
+    }
+
+    persistedWorkspace.withTombstone(true);
+    configRepository.writeStandardWorkspace(persistedWorkspace);
   }
 
   public WorkspaceRead getWorkspace(WorkspaceIdRequestBody workspaceIdRequestBody)
       throws JsonValidationException, IOException, ConfigNotFoundException {
-    return buildWorkspaceReadFromId(workspaceIdRequestBody.getWorkspaceId());
+    final UUID workspaceId = workspaceIdRequestBody.getWorkspaceId();
+    final StandardWorkspace workspace = configRepository.getStandardWorkspace(workspaceId);
+
+    if (workspace.getTombstone()) {
+      throw new NotFoundException("Could not find workspace");
+    }
+
+    return buildWorkspaceRead(workspace);
   }
 
   @SuppressWarnings("unused")
   public WorkspaceRead getWorkspaceBySlug(SlugRequestBody slugRequestBody) throws JsonValidationException, IOException, ConfigNotFoundException {
     // for now we assume there is one workspace and it has a default uuid.
-    return buildWorkspaceReadFromId(PersistenceConstants.DEFAULT_WORKSPACE_ID);
+    final StandardWorkspace workspace = getWorkspaceBySlug(slugRequestBody.getSlug());
+    return buildWorkspaceRead(workspace);
+  }
+
+  private StandardWorkspace getWorkspaceBySlug(String slug)
+      throws JsonValidationException, IOException, KnownException {
+    for (StandardWorkspace workspace : configRepository.listStandardWorkspaces()) {
```

Comment:
Same idea for the prototype.

Reply:
Instead actually. What about adding a configRepository::getStandardWorkspaceBySlug (with the tombstone boolean as well)